tools/tests/test_extensions.py

```diff
@@ -95,16 +95,16 @@ def test_3_example_extension(self):
         # Now that an extension has started, check extension list
         result = test_base.expect(em.extensions, 1)
         self.assertEqual(len(result), 1)
-        ex_uuid = result.keys()[0]
-        ex_data = result[ex_uuid]
+        for ex_uuid in result:
+            ex_data = result[ex_uuid]
```

Comment:
What made necessary this change?
Also shouldn't we just take the first element?

Reply:
I think this is a python2 -> 3 change.
`result.keys()` becomes a 'dict_keys' object which does not support indexing.
Also, the first element is ensured with the preceding test: ` self.assertEqual(len(result), 1)`

Reply:
Shouldn't it then become
```
next(iter(result))
```
It's just that's strange to use a `for` for a one element dictionary, to take one element ^^'.

Reply:
Ah yes, makes sense. 
Anyways, I converted `dict_keys` to `list` and got rid of the loop.